xtext-website/_posts/releasenotes/2024-05-28-version-2-35-0.md

```diff
@@ -40,14 +50,43 @@ String s = "a multi\r\nline string";
+
+This mechanism assumes that you implement the `JvmModelInferrer` "correctly".
+It only checks the first inferred `JvmGenericType` for the same DSL element (i.e., if for an element `Entity` you infer two `JvmGenericType`s, `t1` and `t2`, only the first one will be checked).
+Moreover, it only checks Jvm model elements with an associated source element.
+Concerning intended classes to extend and interfaces to extend/implement, it assumes the model inferrer uses the new `JvmTypesBuilder#setSuperClass(JvmDeclaredType, JvmTypeReference)` and `JvmTypesBuilder#addSuperInterface(JvmDeclaredType, JvmTypeReference)`, respectively.
+
+Currently, this validator must be enabled explicitly through the `composedCheck` in the MWE2 file or the `@ComposedChecks` annotation in the validator, e.g., `@ComposedChecks(validators = JvmGenericTypeValidator.class)`.
+ 
+The Domainmodel example now uses this validator.
+The Xtend validator has been refactored to also use this validator.
+
+### Maven building
+
+When using `xtext-maven-plugin` for Xbase languages, relative paths (instead of absolute paths) are now generated in the `._trace` files.
+
+The configuration option `writeStorageResources` has been added to `xtext-maven-plugin` to write the semantic model, the resource description, and optionally the node model to a ResourceStorage (`.bin` files).
+See [this integration test](https://github.com/eclipse/xtext/tree/main/org.eclipse.xtext.maven.plugin/src/test/resources/it/generate/simple-lang-write-storage-resources) for an example.
+
+### Runtime Testing
 
 The `AbstractFormatterTest` base class has been added to the `org.eclipse.xtext.testing` package to provide a convenient way to test the formatter capabilities. The Xtext Domainmodel example project and the Xtend code base have been extended by concrete formatter test cases to demonstrate the usage of this framework class.
 
+The generated `InjectorProvider` for runtime tests has been improved to make customizations easier w.r.t. OSGI/Maven environments (see [https://github.com/eclipse/xtext/pull/3042](https://github.com/eclipse/xtext/pull/3042)).
+
+The base class `AbstractXtextTests`, for "old"-style testing, is now part of `org.eclipse.xtext.testing`. (See also the removal notice below.)
+
+Two new methods were added `org.eclipse.xtext.xbase.testing.CompilationTestHelper.Result.assertNoErrors()`/`assertNoIssues()` for checking whether during the compilation of the input sources errors, respectively, errors or warnings, were detected.
+
 ### UI Testing
 
 The `AbstractContentAssistTest` class has been extended by API methods to provide a convenient way to test proposals from several resources. The `StatemachineContentAssistTest` test class of the Xtext Statemachine example project demonstrates the usage of these new API methods.
 
+
```

Comment:
can you add new MWE 2.18.0. GSON 2.11 and guava 33.2  and lsp4j 0.23.x here?

Reply:
@cdietrich done: https://github.com/eclipse/xtext-website/pull/47/commits/0a09a19e9d6cf01a6e86249c335d52c2bbbac5be